pygeos/_geometry.pyx

```diff
@@ -217,6 +217,15 @@ cdef _deallocate_arr(void* handle, np.intp_t[:] arr, Py_ssize_t last_geom_i):
 @cython.boundscheck(False)
 @cython.wraparound(False)
 def collections_1d(object geometries, object indices, int geometry_type = 7):
```

Comment:
```suggestion
def collections_1d(np.ndarray[object, ndim=1] geometries, object indices, int geometry_type = 7):
```

(I assume telling cython that it's an numpy array will still give better indexing performance, compared to specifying it as a generic python object)

Reply:
Ah, there is still the `geometries = np.asarray(geometries, dtype=object)` below, so you can also type it there. 

But I think adding the type in the signature is fine, for the low level cython function it is fine to only accept arrays, I think)

Reply:
I went a bit down this road. It does give you more readable code, but also, you can get segfaults from the low-level functions. 

As everything is nicely inplace now (with the `asarray` inside the cython function + checks on dimensionality) I think I rather leave the type coersion inside the function.